src/app/models/constraint-models/combination-constraint.ts

```diff
@@ -107,4 +106,8 @@ export class CombinationConstraint extends Constraint {
       return new TrueConstraint();
     }
   }
+
+  private get modifiesLowerLevels(): boolean {
```

Comment:
can there be a better name for this function?

Reply:
true, I should stick to the 'dimension' naming

Reply:
also the naming for boolean getters normally follows: isFooBar()